### PR TITLE
Run phpstan locally with `lnms dev:check lint`

### PR DESCRIPTION
**Pull request by murrant**
Separate result cache for deprecated functions so we don't break the cache every time.

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Jellyfrog** [comment]
This makes phpstan run twice. Ci helper + seperate linter action

**Comment by murrant** [comment]
@Jellyfrog that sir is why this is a draft :)

**Comment by Jellyfrog** [comment]
> @Jellyfrog that sir is why this is a draft :)
> 
> (edit: now, oops)

;)

File: phpstan-deprecated.neon [hunk 1]
```diff
@@ -6,6 +6,7 @@ parameters:
 
     customRulesetUsed: true
     reportUnmatchedIgnoredErrors: false
+    resultCachePath: %tmpDir%/resultCache-deprecated.php
```
Comment (Jellyfrog) [inline phpstan-deprecated.neon:9]: Good one!

File: LibreNMS/Util/CiHelper.php [hunk 2]
```diff
@@ -273,6 +274,12 @@ public function checkLint()
             $php_lint_cmd = array_merge($php_lint_cmd, $files);
 
             $return += $this->execute('PHP lint', $php_lint_cmd);
+
+            if (! $this->flags['lint_skip_phpstan']) {
+                $phpstan_cmd = [$this->checkPhpExec('phpstan'), 'analyze', '--no-interaction', '--error-format=github',  '--memory-limit=2G'];
```
Comment (Jellyfrog) [inline LibreNMS/Util/CiHelper.php:279]: Better output style for humans
```suggestion
                $phpstan_cmd = [$this->checkPhpExec('phpstan'), 'analyze', '--no-interaction',  '--memory-limit=2G'];
```

Comment (murrant) [reply LibreNMS/Util/CiHelper.php]: Indeed :)

**Review by Jellyfrog — APPROVED** [review]

**Pull request merged**